src/lexer.coffee

```diff
@@ -1033,14 +1033,17 @@ exports.Lexer = class Lexer
   # Compensate for the things we strip out initially (e.g. carriage returns)
   # so that location data stays accurate with respect to the original source file.
   getLocationDataCompensation: (start, end) ->
-    compensation = 0
+    totalCompensation = 0
     initialEnd = end
-    for index, length of @locationDataCompensations
-      index = parseInt index, 10
-      continue unless start <= index and (index < end or index is end and start is initialEnd)
-      compensation += length
-      end += length
-    compensation
+    current = start
+    while current <= end
```

Comment:
This just changes the iteration when calculating the location data compensations for a given `start`/`end` range to iterate on `start..end` instead of iterating on the entire `@locationDataCompensations` data structure every time